### PR TITLE
Added custom names and colors to `Ensemble` class and ensemble selector component

### LINKED ISSUE #552: Global ensemble colors
### Details
- Make it possible to select colors per ensemble in `Ensemble Selector`
- Persist the selection (first iteration: local storage, later: server side)

### PR DESCRIPTION
**Pull request by rubenthoms**
Also...
- makes `Workbench` a constant variable outside of React.
- fixes several smaller issues
- adds an loading overlay when loading ensembles
- adds color tile to ensemble selection components
- adjusted `SimulationTimeSeries` and `SimulationTimeSeriesMatrix` to use custom ensemble colors and names

Closes #552.

### CODE REVIEW

**Review by jorgenherje — CHANGES_REQUESTED** [review]
Really nice work, looks very good! 👍

Just one minor comment.

And a general thought, which does not need to be resolved at the moment, but can be revisited in the future:

With color and display name as a part of the `Ensemble` class, the changes are propagated together with the rest of the Ensemble-info. Thereby editing of these settings in the Ensemble dialog will result in the Ensembles being loaded and processed before new color and custom name is propagated to modules,  without any changes in the rest of the `Ensemble` data. As this is configurable/editable settings, should we find a way to propagate these selections without triggering full loading/fetching of ensembles?

File: frontend/src/lib/components/ColorSelect/colorSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    colors?: string[];
+};
+
+export function ColorSelect(props: ColorSelectProps): JSX.Element {
+    const [selectedColor, setSelectedColor] = React.useState(props.value);
+    const [prevSelectedColor, setPrevSelectedColor] = React.useState(props.value);
+
+    const inputRef = React.useRef<HTMLInputElement>(null);
+
+    if (props.value !== prevSelectedColor) {
```
[the remark this answers is not in the document]
Comment (rubenthoms) [reply frontend/src/lib/components/ColorSelect/colorSelect.tsx:20]: You mean to update it by actually not changing `props.value` but wanting it to override `selectedColor` anyway? The concept often used is controlled (`value` and `onValueChange` set/handled on the outside of the component, e.g. the parent) vs. uncontrolled (only an initial value is given and after mounting the component handles its state by itself and it's only read on the outside) components. Many components offer both by having an `initialValue`, `value`, and an `onChange` prop (but they complain when you set both, `value` and `initialValue`, as you seem undecided if you want to have the component uncontrolled or controlled). Here, we are also offering both, but we do not explicitly ask for an initial value. You could use it as an uncontrolled component by just giving the initial value once (as `value`) and never changing it or an controlled component by handling/storing the component's state in the parent as well (in sync with the component) and giving it in as a prop (if a new `props.value` is different from the last selection (both `props.value`, `selectedColor`, and `prevSelectedColor` are synced), the component's selection is adjusted accordingly). However, this solution also makes it possible to change the current selection of an uncontrolled component by changing the initial value via `props.value` as long as it is different from the last initial value (one could of course argue that there aren't many use cases for this scenario). Coming back to your question: It is possible to handle your use case by using the component as a controlled component. If it is used as an uncontrolled one, you are of course right, the update would render ineffective as you would reapply the same initial value.

[the remark this answers is not in the document]
Comment (rubenthoms) [reply frontend/src/lib/components/ColorSelect/colorSelect.tsx:20]: You mean to update it by actually not changing `props.value` but wanting it to override `selectedColor` anyway? The concept often used is controlled (`value` and `onValueChange` set/handled on the outside of the component, e.g. the parent) vs. uncontrolled (only an initial value is given and after mounting the component handles its state by itself and it's only read on the outside) components. Many components offer both by having an `initialValue`, `value`, and an `onChange` prop (but they complain when you set both, `value` and `initialValue`, as you seem undecided if you want to have the component uncontrolled or controlled). Here, we are also offering both, but we do not explicitly ask for an initial value. You could use it as an uncontrolled component by just giving the initial value once (as `value`) and never changing it or an controlled component by handling/storing the component's state in the parent as well (in sync with the component) and giving it in as a prop (if a new `props.value` is different from the last selection (both `props.value`, `selectedColor`, and `prevSelectedColor` are synced), the component's selection is adjusted accordingly). However, this solution also makes it possible to change the current selection of an uncontrolled component by changing the initial value via `props.value` as long as it is different from the last initial value (one could of course argue that there aren't many use cases for this scenario). Coming back to your question: It is possible to handle your use case by using the component as a controlled component. If it is used as an uncontrolled one, you are of course right, the update would render ineffective as you would reapply the same initial value.

**Comment by rubenthoms** [comment]
> Really nice work, looks very good! 👍
> 
> Just one minor comment.
> 
> And a general thought, which does not need to be resolved at the moment, but can be revisited in the future:
> 
> With color and display name as a part of the `Ensemble` class, the changes are propagated together with the rest of the Ensemble-info. Thereby editing of these settings in the Ensemble dialog will result in the Ensembles being loaded and processed before new color and custom name is propagated to modules, without any changes in the rest of the `Ensemble` data. As this is configurable/editable settings, should we find a way to propagate these selections without triggering full loading/fetching of ensembles?

Of course, as already mentioned in our conversation yesterday, this is not optimal yet as all ensembles are loaded again when anything in the ensemble selector is changed and a new ensemble set is created. This just made it easier to achieve an update of all modules using the current ensemble set. Moreover, I assume that there aren't very frequent changes to the ensemble names/colors and the ensemble data should be stored in the Tanstack query cache for 60s (and later also in the browser's cache), making this whole operation rather fast in most of the cases. 
However, I totally agree that we should aim to implement another method for updating modules on ensemble set color/name changes and avoid re-fetching all ensemble data in the future.

**Review by jorgenherje — APPROVED** [review]
👍

**Pull request merged**